File: excellence/services/api/models/extraction_job.py
from typing import List, Literal, Optional, Union

from .base import CamelModel
from .extraction_result import ExtractionResult


InputModality = Literal["text", "image", "url", "mixed"]
ExtractionPhase = Literal["phase1", "phase2", "phase3"]
JobStatus = Literal[
    "queued",
    "processing",
    "awaiting_preview",
    "committed",
    "failed",
    "cancelled",
]


class TextInput(CamelModel):
    type: Literal["text"]
    content: str


class ImageInput(CamelModel):
    type: Literal["image"]
    base64: str
    mime_type: Literal["image/png", "image/jpeg", "image/webp"]


class UrlInput(CamelModel):
    type: Literal["url"]
    url: str
    table_hint: Optional[str] = None


class MixedInput(CamelModel):
    type: Literal["mixed"]
    inputs: List[Union[TextInput, ImageInput, UrlInput]]


ExtractionInput = Union[TextInput, ImageInput, UrlInput, MixedInput]


class ExtractionError(CamelModel):
    code: str
    message: str
    recoverable: bool


class ExtractionJob(CamelModel):
    id: str
    modality: InputModality
    phase: ExtractionPhase
    status: JobStatus
    priority: int
    input: ExtractionInput
    result: Optional[ExtractionResult] = None
    error: Optional[ExtractionError] = None
    created_at: int
    updated_at: int

File: excellence/packages/extraction-core/src/models/ExtractionJob.ts
export type InputModality = 'text' | 'image' | 'url' | 'mixed';
export type ExtractionPhase = 'phase1' | 'phase2' | 'phase3';
export type JobStatus =
	| 'queued'
	| 'processing'
	| 'awaiting_preview'
	| 'committed'
	| 'failed'
	| 'cancelled';

export interface TextInput {
	type: 'text';
	content: string;
}

export interface ImageInput {
	type: 'image';
	base64: string;
	mimeType: 'image/png' | 'image/jpeg' | 'image/webp';
}

export interface UrlInput {
	type: 'url';
	url: string;
	tableHint?: string;
}

export interface MixedInput {
	type: 'mixed';
	inputs: Array<TextInput | ImageInput | UrlInput>;
}

export type ExtractionInput = TextInput | ImageInput | UrlInput | MixedInput;

export interface ExtractionJob {
	id: string;
	modality: InputModality;
	phase: ExtractionPhase;
	status: JobStatus;
	priority: number;
	input: ExtractionInput;
	result?: import('./ExtractionResult').ExtractionResult;
	error?: ExtractionError;
	createdAt: number;
	updatedAt: number;
}

export interface ExtractionError {
	code: string;
	message: string;
	recoverable: boolean;
}
